src/cogent3/core/sequence.py

```diff
@@ -1209,11 +1209,24 @@ def copy_annotations(self, seq_db: SupportsFeatures) -> None:
 
         self.annotation_db.update(seq_db, seqids=self.name)
 
-    def copy(self, exclude_annotations=False):
-        """returns a copy of self"""
-        offset = self._seq.offset + self._seq.start
+    def copy(self, exclude_annotations: bool = False, sliced: bool = True):
+        """returns a copy of self
+
+        Parameters
+        -----------
+        sliced
+            Slices underlying sequence with start/end of self coordinates. The offset
+            is retained.
+        exclude_annotations
+            drops annotation_db when True
+        """
+        annotation_offset = self.annotation_offset if sliced else self._seq.offset
```

Comment:
Correct, aside from the second point. `Sequence.annotation_offset` returns the new `SeqView.parent_start` because the sum of offset and start can be negative when it's a reversed view.

Assigning to `Sequence.annotation_offset` is to represent external knowledge provided by the user. That get's propagated to the `SeqView` instance. Slicing operations on the SeqView don't change its offset, but it's relative start.

Hope that helps

Reply:
thanks 🙂